File: logistics-informatization/mrli-core/mrli-pro/ykport-bshuisystem/ykport-bshuisystem-api/src/main/java/com/newweb/mrli/ykport/baoshuisystem/entity/YkbfSingClass.java
/*
 *    Copyright (c) 2018-2025, mrli All rights reserved.
 *
 * Redistribution and use in source and binary forms, with or without
 * modification, are permitted provided that the following conditions are met:
 *
 * Redistributions of source code must retain the above copyright notice,
 * this list of conditions and the following disclaimer.
 * Redistributions in binary form must reproduce the above copyright
 * notice, this list of conditions and the following disclaimer in the
 * documentation and/or other materials provided with the distribution.
 * Neither the name of the pig4cloud.com developer nor the names of its
 * contributors may be used to endorse or promote products derived from
 * this software without specific prior written permission.
 * Author: mrli
 */

package com.newweb.mrli.ykport.baoshuisystem.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import com.baomidou.mybatisplus.extension.activerecord.Model;
import com.newweb.mrli.ykport.baoshuisystem.utils.ExcelUtil;
import io.swagger.annotations.ApiModel;
import io.swagger.annotations.ApiModelProperty;
import lombok.Data;
import lombok.EqualsAndHashCode;
import java.io.Serializable;
import java.time.LocalDateTime;

/**
 * 单个工班查询
 *
 * @author pigx code generator
 * @date 2021-10-24 15:49:10
 */
@Data
@TableName("ykport1.YKBF_SING_CLASS")
@EqualsAndHashCode(callSuper = true)
@ApiModel(value = "单个工班查询")
public class YkbfSingClass extends Model<YkbfSingClass> {

    private static final long serialVersionUID = 1L;

    /**
     * 代码
     */
    @ApiModelProperty(value="代码")
    private String lxdm;

    /**
     * 货票号
     */
    @ApiModelProperty(value="货票号")
    private String bh;

    /**
     * 货名
     */
	@ExcelUtil.ExcelAttribute(column = "2",name = "品名")
    @ApiModelProperty(value="货名")
    private String hm;

    /**
     * 提单号
     */
    @ApiModelProperty(value="提单号")
    private String tdh;

    /**
     * 船名
     */
    @ApiModelProperty(value="船名")
    private String cm;

    /**
     * 计划量
     */
    @ApiModelProperty(value="计划量")
    private String jhl;

    /**
     * 完成车
     */
    @ApiModelProperty(value="完成车")
    private String csl;

    /**
     * 完成量
     */
	@ExcelUtil.ExcelAttribute(column = "2",name = "出库量")
    @ApiModelProperty(value="完成量")
    private String wcl;

    /**
     * 剩余量
     */
    @ApiModelProperty(value="剩余量")
    private String syl;

    /**
     * 本班车
     */
    @ApiModelProperty(value="本班车")
    private String benc;

    /**
     * 未回车
     */
    @ApiModelProperty(value="未回车")
    private String weic;

    /**
     * 本班货重
     */
    @ApiModelProperty(value="本班货重")
    private String benhz;

    /**
     * 称别组
     */
    @ApiModelProperty(value="称别组")
    private String cbs;

    /**
     * 申请单位
     */
    @ApiModelProperty(value="申请单位")
    private String sqdw;

    /**
     * 班次
     */
	@ExcelUtil.ExcelAttribute(column = "3",name = "日期")
    @ApiModelProperty(value="班次")
    private String rqcbs;

	/**
	 * 班次
	 */
	@ApiModelProperty(value="班次")
	@TableField(exist = false)
	private String[] startDte;

}

File: logistics-informatization/mrli-ui/src/api/baoshuisystem/ykbfsingclass.js
/*
 *    Copyright (c) 2018-2025, mrli All rights reserved.
 *
 * Redistribution and use in source and binary forms, with or without
 * modification, are permitted provided that the following conditions are met:
 *
 * Redistributions of source code must retain the above copyright notice,
 * this list of conditions and the following disclaimer.
 * Redistributions in binary form must reproduce the above copyright
 * notice, this list of conditions and the following disclaimer in the
 * documentation and/or other materials provided with the distribution.
 * Neither the name of the pig4cloud.com developer nor the names of its
 * contributors may be used to endorse or promote products derived from
 * this software without specific prior written permission.
 * Author: mrli
 */

import request from "@/router/axios";

export function fetchList(query) {
  return request({
    url: "/baoshuisystem/ykbfsingclass/page",
    method: "get",
    params: query
  });
}
export function getAll(query) {
  return request({
    url: "/baoshuisystem/ykbfsingclass/noPage",
    method: "get",
    params: query
  });
}
export function getObj(id) {
  return request({
    url: "/baoshuisystem/ykbfsingclass/" + id,
    method: "get"
  });
}

export function printToExcel(obj) {
  return request({
    url: "/baoshuisystem/ykbfsingclass/printToExcel",
    method: "post",
    data: obj,
    responseType: "arraybuffer"
  }).then(response => {
    // 处理返回的文件流
    const blob = new Blob([response.data], { type: "application/zip " });
    const link = document.createElement("a");
    link.href = URL.createObjectURL(blob);
    link.download = "日操作量表.xls";
    document.body.appendChild(link);
    link.click();
    window.setTimeout(function() {
      URL.revokeObjectURL(blob);
      document.body.removeChild(link);
    }, 0);
  });
}



export function exportList(obj) {
  return request({
    url: "/baoshuisystem/ykbfsingclass/exportList",
    method: "post",
    data: obj,
    responseType: "arraybuffer"
  }).then(response => {
    // 处理返回的文件流
    const blob = new Blob([response.data], { type: "application/zip " });
    const link = document.createElement("a");
    link.href = URL.createObjectURL(blob);
    link.download = "单个工班查询管理.xls";
    document.body.appendChild(link);
    link.click();
    window.setTimeout(function() {
      URL.revokeObjectURL(blob);
      document.body.removeChild(link);
    }, 0);
  });
}